web/src/scheduled_messages.js

```diff
@@ -141,6 +141,8 @@ export function initialize() {
     if (scheduled_messages_data.length === 0) {
         scheduled_messages_data = page_params.scheduled_messages;
     } else {
+        // We received an `add` event before `scheduled_messages_data` was initialized,
+        // so, we need to just add the old scheduled messages here.
```

Comment:
How would that be possible? This gets initialized before `server_events.js` does. I think you can just cut this if statement -- and also do the pattern of popping `page_params.scheduled_messages` in `ui_init.js` and just passing the value here.

Also, I just noticed, `dispatch_normal_events`'s blocks aren't sorted; can you fix that?

Reply:
Sorry, fixed all of that.